### PR TITLE
Don't call kinit cmd from rd_kafka_new()

### LINKED ISSUE #2912: Random latency spikes when producing messages
Description
===========
I have been running latency tests for my application using librdkafka and noticed some random latency spikes of upto 1 second when producing messages. This happens even when my queue.buffering.max.ms is only set to 10 milliseconds. I am trying to investigate if the issue is within librdkafka, or something specific to my kafka configuration or at the server.

How to reproduce
================
The issue can be reproduced by running kafka server locally on Mac, and running examples/rdkakfa_performance, setting two extra configurations:
```
rd_kafka_conf_set(conf, "queue.buffering.max.ms", "10", NULL, 0);
rd_kafka_conf_set(conf, "request.required.acks", "1", NULL, 0);
```

```
./examples/rdkafka_performance -t test-topic -b localhost:9092 -P -s 100 -r 1000 -l -A latency.txt
```
This issue can be reproduced more often by producing at a rate of 1k msgs/sec. As this happens at random, it has to be running for a few minutes before we see any spikes in latency.

Checklist
=========
 - [x] librdkafka version (release number or git tag): `1.4.2`
 - [x] Apache Kafka version: `2.5.0`
 - [x] librdkafka client configuration: `queue.buffering.max.ms=10, request.required.acks=1`
 - [x] Operating system: `macOs Mojave`
 - [ ] Provide logs (with `debug=..` as necessary) from librdkafka
 - [ ] Provide broker log excerpts
 - [ ] Critical issue

Below are some sample logs from running examples/rdkafka_performance. For a produce rate of 1k/sec, and queue buffering of 10ms, we should be seeing max of 10 messages in queue. However, some times, the queue has 100s of messages.

```
$ ./examples/rdkafka_performance -t test-topic -b localhost:9092 -P -s 100 -r 1000 -l -A latency.txt
% Sending messages of size 100 bytes
% 833 messages produced 828 delivered, 5 in queue
% 1681 messages produced 1676 delivered, 5 in queue
% 2511 messages produced 2508 delivered, 3 in queue
% 3311 messages produced 2788 delivered, 523 in queue
% 4132 messages produced 4127 delivered, 5 in queue
% 4963 messages produced 4961 delivered, 2 in queue
% 5815 messages produced 5810 delivered, 5 in queue
% 6654 messages produced 6649 delivered, 5 in queue
% 7462 messages produced 6964 delivered, 498 in queue
% 8273 messages produced 7864 delivered, 409 in queue
% 9107 messages produced 9104 delivered, 3 in queue
% 9959 messages produced 9954 delivered, 5 in queue
% 10795 messages produced 10789 delivered, 6 in queue
% 11621 messages produced 11618 delivered, 3 in queue
% 12404 messages produced 11892 delivered, 512 in queue
% 13230 messages produced 13221 delivered, 9 in queue
% 14048 messages produced 13424 delivered, 624 in queue
% 14888 messages produced 14882 delivered, 6 in queue
% 15730 messages produced 15726 delivered, 4 in queue
% 16559 messages produced 16442 delivered, 117 in queue
% 17390 messages produced 17383 delivered, 7 in queue
% 18217 messages produced 17751 delivered, 466 in queue
% 19048 messages produced 19041 delivered, 7 in queue
% 19875 messages produced 19867 delivered, 8 in queue
% 20706 messages produced 20698 delivered, 8 in queue
% 21534 messages produced 21527 delivered, 7 in queue
% 22349 messages produced 21612 delivered, 737 in queue
% 23179 messages produced 23171 delivered, 8 in queue
% 23551 messages produced 23551 delivered, 0 in queue
```

Per message top latencies reach above 900 milliseconds:
```
$ cat latency.txt | sort -n -r | head -10
945274
943359
941492
940687
940258
939312
936452
935604
935409
932852
```

I also verified that this is not just because server is not able to handle the load, by running at peak rate:
```
$ ./examples/rdkafka_performance -t test-topic -b localhost:9092 -P -s 100
% Sending messages of size 100 bytes
% 1335399 messages produced (133539900 bytes), 1309811 delivered (offset 7433195, 0 failed) in 1002ms: 1305974 msgs/s and 130.60 MB/s, 0 produce failures, 25589 in queue, no compression
% 2675854 messages produced (267585400 bytes), 2659421 delivered (offset 7769786, 0 failed) in 2003ms: 1327255 msgs/s and 132.73 MB/s, 4 produce failures, 16435 in queue, no compression
```

### Debugging internals
I tried debugging the rdkafka code, and noticed the major lag is happening at https://github.com/edenhill/librdkafka/blob/master/src/rdkafka_transport.c#L965 in poll(). However, I do not understand what that means, and also if that is the main cause of delay.

### PR DESCRIPTION
**Pull request by edenhill**
I think the broker wakeups might also fix #2912

### CODE REVIEW

File: tests/0081-admin.c [hunk 1]
```diff
@@ -1708,6 +1708,8 @@ static void do_test_DeleteConsumerGroupOffsets (const char *what,
                                   NULL, 0,
                                   15*1000);
 
+        rd_sleep(1); /* Additional wait time for cluster propagation */
```
Comment (jliunyu) [inline tests/0081-admin.c:1711]: How long will it take for the cluster propagation usually? I'm wondering that will 1 additional second long enough?

Comment (edenhill) [reply tests/0081-admin.c:1711]: Fair question, and the answer is we can't really know.
On the line above we query the cluster for metadata on the topic, but that only represents a single broker's view, while other brokers may not yet have the full info.
The proper solution would be to query each broker in the cluster, but there's no easy way to do that with librdkafka.

So this arbitrary 1 second sleep is a naiive but cheap workaround that improves test stability.

**Review by jliunyu — COMMENTED** [review]
The change looks good to me! Since I'm not very familiar with the code yet, I will let Matt to give the final approval.

File: src/rdkafka_request.c [hunk 2]
```diff
@@ -4526,6 +4526,8 @@ ut_create_msgs (rd_kafka_msgq_t *rkmq, uint64_t msgid, int cnt) {
 
                 rkm = ut_rd_kafka_msg_new(0);
                 rkm->rkm_u.producer.msgid = msgid++;
+                rkm->rkm_ts_enq = rd_clock();
+                rkm->rkm_ts_timeout = rkm->rkm_ts_enq + (900 * 1000 * 1000);
```
Comment (mhowlett) [inline src/rdkafka_request.c:4530]: hard coded constant?

Comment (mhowlett) [reply src/rdkafka_request.c:4530]: oh, it's a unit test

**Review by mhowlett — APPROVED** [review]

**Pull request merged**